src/review/review.controller.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Param,
+  Patch,
+  Post,
+} from '@nestjs/common';
+import { ReviewService } from './review.service';
+import { CreateReviewDto } from './dto/create-review.dto';
+import { UpdateReviewDto } from './dto/update-review.dto';
+import { CurrentUser, User } from '@app/common';
+import {
+  ApiTags,
+  ApiOperation,
+  ApiResponse,
+  ApiBody,
+  ApiParam,
+} from '@nestjs/swagger';
+
+@ApiTags('reviews')
+@Controller('reviews')
+export class ReviewController {
+  constructor(private readonly reviewService: ReviewService) {}
+
+  @Post()
+  @HttpCode(HttpStatus.CREATED)
+  @ApiOperation({ summary: 'Create a new review' })
+  @ApiResponse({
+    status: HttpStatus.CREATED,
+    description: 'Review successfully created',
+  })
+  @ApiResponse({
+    status: HttpStatus.BAD_REQUEST,
+    description: 'Invalid input data',
+  })
+  @ApiBody({ type: CreateReviewDto })
+  async create(
+    @Body() createReviewDto: CreateReviewDto,
+    @CurrentUser() { userId }: User,
+  ) {
+    return this.reviewService.create(createReviewDto, userId);
+  }
+
+  @Get('guide/:guideId')
+  @HttpCode(HttpStatus.OK)
+  @ApiOperation({ summary: 'Get all reviews of a guide by guide id' })
+  @ApiResponse({ status: HttpStatus.OK, description: 'List of all reviews' })
+  async findAll(@Param('guideId') guideId: string) {
+    return this.reviewService.findAll(+guideId);
+  }
+
```

Comment:
This shouldnt be here. Once you get the guide from its controller you should also get the reviews.

Reply:
Normally I would agree with you but the thing is, as I mentioned in the front-end PR, is that in that way I cant get the user's relationship from review